test/test-lex.cpp

```diff
@@ -161,13 +161,120 @@ TEST(test_lex, lex_binary_numbers) {
   check_single_token(u8"0B010101010101010", token_type::number);
 }
 
+// fails
```

Comment:
Nit: Remove this comment. Preferably, leave this test for a separate commit. But I won't get mad if you leave the test here.

Reply:
Added more specific TODO